### DISCUSSION TITLE
Slow Nextjs site according to PageSpeed Insights

### DISCUSSION THREAD

**Discussion by nonoumasy** [vercel/next.js#42536] [Help]
Hello, I'm trying to optimize the homepage on my site: https://history-maps.com/
I'm using Nextjs 12 with MUI and plain CSS.I use the Nextjs Image component set on lazy.

When I inspect the size using:
document.getElementById("__NEXT_DATA__").text

It returns 45kb. 

My issue is that whenever I do use https://pagespeed.web.dev/ on my site for both desktop and mobile, I get a score of 75 for desktop and a score of 28 for mobile. Is it normal for the mobile score to be low compared to the desktop? 

I attached the diagnostics that PSI reports. Can someone pls just give me an overview of something I can do to improve my score.
<img width="999" alt="Screen Shot 2022-11-06 at 4 04 27 PM" src="https://user-images.githubusercontent.com/6312676/200179445-71e9acbc-b957-4c34-b70c-58a11708fe26.png">

is the Avoid an excessive DOM size the main culprit? My homepage is very content heavy in general and the homepage does show all 250 stories, which includes images. I did reduce this to only essential data, but there is still a lot of content.

Any help is appreciated.

**Comment by skinheadz** [comment]
Wow, mapbox that you are using is looking amazing. Answering your question, I always get worse rating/loadspeed on mobile versus desktop, it has to be part of the score calculations. 

I suggest trying @next/bundle-analyzer, it gives you a beautiful output for client side and server side with all the libraries/files/components that you load and their respective size.
Here's how to install:
next.config.js: require it and add module.exports = withBundleAnalyzer(your current config);
run "ANALYZE=true npm run build"
Please see screenshots for details.
And here is a detailer step-by-step [blogpost](https://daily-dev-tips.com/posts/exploring-the-nextjs-bundle-analyzer/)


<img width="458" alt="Screen Shot 2022-11-07 at 4 45 17 AM" src="https://user-images.githubusercontent.com/113230343/200313749-667bed6f-354e-4155-a19d-5bd02d00050c.png">

<img width="1050" alt="Screen Shot 2022-11-07 at 4 39 53 AM" src="https://user-images.githubusercontent.com/113230343/200313428-6243b25b-0411-4248-99b3-63482f15cc84.png">

**Reply by nonoumasy** [reply]
thanks @skinheadz

I installed that package (pretty simple install).
When I analyze, I get this for the client.html

<img width="1440" alt="Screen Shot 2022-11-16 at 2 12 21 PM" src="https://user-images.githubusercontent.com/6312676/202190885-eb970ca3-2e3f-43a6-8c58-28d880af217f.png">

I guess based on this, mapbox, firebase, jsts, and quill are the majority of the initial page load? although I don't really use mapbox on the homepage.

**Comment by Bahlaouane-Hamza** [comment]
Hi,
You will need to patch Next.js for this, here's one for next@12.1.6
You can use https://www.npmjs.com/package/patch-package to auto install the patch.

[next+12.1.6.patch](https://github.com/vercel/next.js/files/11009343/next%2B12.1.6.patch)

For this part
```js
+    delete props1.pageProps.node
+    delete props1.pageProps.i18n
+    delete props1.pageProps.menus
```
I'm removing only the props i don't wanna keep in the HTML, If you are using NextAuth for example, you want to keep `session` coming from SSR, and this may vary depending on your use case.

And finally on the _hydrate part, i just restore the props i removed.

Hope this helps.
